base/linalg/tridiag.jl

```diff
@@ -19,7 +21,14 @@ function SymTridiagonal{Td,Te}(dv::Vector{Td}, ev::Vector{Te})
     SymTridiagonal(convert(Vector{T}, dv), convert(Vector{T}, ev))
 end
 
-SymTridiagonal(A::AbstractMatrix) = diag(A,1)==diag(A,-1)?SymTridiagonal(diag(A), diag(A,1)):throw(DimensionMismatch("matrix is not symmetric; cannot convert to SymTridiagonal"))
+function SymTridiagonal(A::AbstractMatrix)
+    if diag(A,1) == diag(A,-1)
```

Comment:
For me at least, this would be clearer with the throw first, and then what you are returning second?